src/mgcommon.h

```diff
@@ -15,6 +15,41 @@
 #ifndef MGCLIENT_MGCOMMON_H
 #define MGCLIENT_MGCOMMON_H
 
+#ifdef MGCLIENT_ON_APPLE
```

Comment:
have mgcommon.h in apple/linux/windows folders and have conditional includes under ifdef?
#ifdef MGCLIENT_ON_APPLE
#include "apple/mgcommon.h"
#endif

Reply:
Done!